srcs/game/game.js

```diff
@@ -0,0 +1,172 @@
+			else
+				Player.Paddle.y += Player.Paddle.speed;
+		}
+		else
+		{
+			if (Player.Paddle.y == 0)
+				return ;
+			if (Player.Paddle.y - Player.Paddle.speed <= 0)
+				Player.Paddle.y += canvas.height - Player.Paddle.y;
+			else
+				Player.Paddle.y += Player.Paddle.speed;
+		}
+	}
+
+	function checkCollision(paddle) {
+		const closestX = Math.max(paddle.x, Math.min(ball.x, paddle.x + PADDLE_WIDTH));
+		const closestY = Math.max(paddle.y, Math.min(ball.y, paddle.y + PADDLE_HEIGHT));
+		
+		const distanceX = ball.x - closestX;
+		const distanceY = ball.y - closestY;
+		const distance = Math.sqrt(distanceX * distanceX + distanceY * distanceY);
+		
+		return {
+			hasCollision: distance <= ball.radius,
+			isVertical: Math.abs(distanceY) > Math.abs(distanceX),
+			distanceX: distanceX,
+			distanceY: distanceY
+		};
+	}
+
+	function handlePaddleCollision(player) {
+		const collision = checkCollision(player.Paddle);
+		
+		if (collision.hasCollision) {
+			if (collision.isVertical) {
+				ball.speedY *= -1;
+				if (collision.distanceY > 0) {
+					ball.y = player.Paddle.y + PADDLE_HEIGHT + ball.radius;
+				} else {
+					ball.y = player.Paddle.y - ball.radius;
+				}
+			} else {
+				ball.speedX *= -1;
+				let relativeIntersection = (player.Paddle.y + (PADDLE_HEIGHT * 0.5) - ball.y);
+				let normalizedIntersection = relativeIntersection / (PADDLE_HEIGHT * 0.5);
+				ball.speedY = -normalizedIntersection * 5;
+				if (collision.distanceX > 0) {
+					ball.x = player.Paddle.x + PADDLE_WIDTH + ball.radius;
+				} else {
+					ball.x = player.Paddle.x - ball.radius;
+				}
+			}
+		}
+	}
+
+	function updateBall() {
+		ball.x += ball.speedX;
+		ball.y += ball.speedY;
+		if (ball.y - ball.radius < 0 || ball.y + ball.radius > canvas.height)
+			ball.speedY *= -1;
+		if (ball.x - ball.radius < 0 || ball.x + ball.radius > canvas.width)
+		{
+			if (ball.x - ball.radius < 0)
+			{
+				Player2.score++;
+				ball.x = canvas.width * 0.7;
+				ball.y = canvas.height * 0.5;
+			}
+			else
+			{
+				Player1.score++;
+				ball.x = canvas.width * 0.3;
+				ball.y = canvas.height * 0.5;
+			}
+//			ball.speedX *= -1;
+		}
+	}
+	
+	function gameLoop() {
+		ctx.clearRect(0, 0, canvas.width, canvas.height);
+		ctx.font = '48px serif';
+		ctx.textBaseline = 'hanging';
+		ctx.fillStyle = 'white';
+		ctx.fillText(Player1.score + " : " + Player2.score, canvas.width * 0.45, canvas.height * 0.10);
+		updatePlayer(Player1);
+		updatePlayer(Player2);
+		handlePaddleCollision(Player1);
+		handlePaddleCollision(Player2);
+		updateBall();
+		ctx.fillStyle = Player1.color;
+		ctx.fillRect(Player1.Paddle.x, Player1.Paddle.y, PADDLE_WIDTH, PADDLE_HEIGHT);
+		ctx.fillStyle = Player2.color;
+		ctx.fillRect(Player2.Paddle.x, Player2.Paddle.y, PADDLE_WIDTH, PADDLE_HEIGHT);
+		ctx.beginPath();
+		ctx.arc(ball.x, ball.y, ball.radius, 0, 2 * Math.PI);
+		ctx.fillStyle = ball.color;
+		ctx.fill();
+		ctx.closePath();
+		requestAnimationFrame(gameLoop);
+	}
```

Comment:
you should make this function longer, it's too easy to read.

Reply:
>![20241118_214818.jpg](https://github.com/user-attachments/assets/962eaec2-4175-4fe2-b4f3-e6d616560baa)